contrib/storage-hive/core/src/test/java/org/apache/drill/exec/hive/HiveTestConfig.java

```diff
@@ -0,0 +1,214 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.hive;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import org.apache.drill.common.exceptions.ExecutionSetupException;
+import org.apache.drill.exec.server.Drillbit;
+import org.apache.drill.exec.store.StoragePluginRegistry;
+import org.apache.drill.exec.store.hive.HiveStoragePlugin;
+import org.apache.drill.exec.store.hive.HiveStoragePluginConfig;
+import org.apache.drill.shaded.guava.com.google.common.base.Preconditions;
+import org.apache.drill.test.BaseDirTestWatcher;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.conf.HiveConf.ConfVars;
+import org.apache.hadoop.hive.ql.Driver;
+import org.apache.hadoop.hive.ql.session.SessionState;
+
+import static org.apache.drill.exec.hive.HiveTestUtilities.createDirWithPosixPermissions;
+
+
+/**
+ * Class for flexible configuration of Hive tests, it's designed to have
+ * minimal public API but feel free to add methods if you feel that it's
+ * really useful and necessary.
+ * <p>
+ * With this class you're not tied to HiveTestBase or HiveTestDataGenerator anymore
+ * and can generate only specific test data necessary for your test class. Such
+ * strategy allows faster execution of tests comparing to extending of HiveTestBase,
+ * because you don't have to wait while all those data from HiveTestDataGenerator will be set up.
+ * <p>
+ * Example of how this class may be used with ClusterFixture:
+ * <p>
+ * <pre><code>
+ * public class HiveExampleTest extends ClusterTest {
+ *    private static HiveTestConfig hiveTestConfig;
+ *
+ *    {@literal @}BeforeClass
+ *    public static void setUp() throws Exception {
+ *            startCluster(ClusterFixture.builder(dirTestWatcher));
+ *            hiveTestConfig = HiveTestConfig.dirWatcher(dirTestWatcher)
+ *                            .data(HiveExampleTest::generateData)
+ *                            .addHivePluginTo(cluster.drillbit());
+ *    }
+ *
+ *    // may be method whose parameters suitable for Consumer<Driver> or BiConsumer<HiveTestConfig, Driver>
+ *    private static void generateData(Driver driver) {
+ *            // set up data using HiveTestUtilities.executeQuery(driver, sqlQuery);
+ *    }
+ *
+ *    {@literal @}AfterClass
+ *    public static void tearDown() throws Exception {
+ *            hiveTestConfig.removeHivePluginFrom(cluster.drillbit());
```

Comment:
Could you please check if this method will be called when`@BeforeClass` fails and `hiveTestConfig` is not initialized? If yes, then we need null check to avoid NPE.